### PR TITLE
Refactor `handleNativeResponse` for better code reusability

### PR DESCRIPTION
**Pull request by sameerag**
This PR:
* Breaks down `handleNativeResponse` into various blocks
* Cache logic separated

Will be followed up with silent cache lookup updates in the new helper functions

### CODE REVIEW

File: change/@azure-msal-browser-9df1850b-ffe9-4027-969f-dad511cc00f6.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+  "type": "none",
```
Comment (tnorling) [inline change/@azure-msal-browser-9df1850b-ffe9-4027-969f-dad511cc00f6.json]:
```suggestion
  "type": "patch",
```

File: lib/msal-browser/src/interaction_client/NativeInteractionClient.ts [hunk 2]
```diff
@@ -231,25 +231,138 @@ export class NativeInteractionClient extends BaseInteractionClient {
...
+
+    /**
+     * creates accountEntity
+     * @param request 
+     * @param response 
+     * @param homeAccountIdentifier 
+     * @param idTokenObj 
+     * @returns 
+     */
+    protected createAccountEntity(response: NativeResponse, homeAccountIdentifier: string, idTokenObj: AuthToken, authority: string): AccountEntity {
```
Comment (tnorling) [inline lib/msal-browser/src/interaction_client/NativeInteractionClient.ts:293]: Not sure I see the point of all these individual one line helpers?

File: lib/msal-browser/src/interaction_client/NativeInteractionClient.ts [hunk 3]
```diff
@@ -328,33 +440,27 @@ export class NativeInteractionClient extends BaseInteractionClient {
         this.nativeStorageManager.setIdTokenCredential(idTokenEntity);
 
         // cache accessToken in inmemory storage
-        const expiresIn: number = (responseTokenType === AuthenticationScheme.POP)
+        const expiresIn: number = (request.tokenType === AuthenticationScheme.POP)
```
Comment (tnorling) [inline lib/msal-browser/src/interaction_client/NativeInteractionClient.ts:432]: Why was this changed from response to request?

File: lib/msal-browser/src/interaction_client/NativeInteractionClient.ts (hunk 2 above)
Comment (peterzenz) [reply lib/msal-browser/src/interaction_client/NativeInteractionClient.ts:293]: For 1p overloads?

File: lib/msal-browser/src/interaction_client/NativeInteractionClient.ts (hunk 3 above)
Comment (sameerag) [reply lib/msal-browser/src/interaction_client/NativeInteractionClient.ts:432]: It is always `request.tokenType`, was assigned to `responseTokenType` in the POP token generation logic.

File: lib/msal-browser/src/interaction_client/NativeInteractionClient.ts (hunk 2 above)
Comment (sameerag) [reply lib/msal-browser/src/interaction_client/NativeInteractionClient.ts:293]: We are calling them in the 1P PR. Wanted to have a helper to not repeat it.

File: change/@azure-msal-browser-9df1850b-ffe9-4027-969f-dad511cc00f6.json (hunk 1 above)
Comment (sameerag) [reply change/@azure-msal-browser-9df1850b-ffe9-4027-969f-dad511cc00f6.json]: Well, is it really a patch? We are not fixing a bug, it is purely code refactor.

File: lib/msal-browser/src/interaction_client/NativeInteractionClient.ts [hunk 4]
```diff
@@ -231,25 +231,138 @@ export class NativeInteractionClient extends BaseInteractionClient {
...
+    /**
+     * 
+     * @param request 
+     * @param response 
+     * @param accountEntity 
+     * @param idTokenObj 
+     * @param authority 
+     * @param uid 
+     * @param tid 
+     * @param responseScopes
```
Comment (peterzenz) [inline lib/msal-browser/src/interaction_client/NativeInteractionClient.ts]: Can you check your input paremeters for the method against the comments here?  Some of these don't line up or are missing in different orders in this change.

**Review by peterzenz — APPROVED** [review]
Please check the doc/comments for correctness before committing.

File: lib/msal-browser/src/interaction_client/NativeInteractionClient.ts (hunk 4 above)
Comment (sameerag) [reply lib/msal-browser/src/interaction_client/NativeInteractionClient.ts]: Addressed the comments, they are outdated. Thanks for catching this.

File: change/@azure-msal-browser-9df1850b-ffe9-4027-969f-dad511cc00f6.json (hunk 1 above)
Comment (tnorling) [reply change/@azure-msal-browser-9df1850b-ffe9-4027-969f-dad511cc00f6.json]: Any change to published code is not a `none` change

Comment (sameerag) [reply change/@azure-msal-browser-9df1850b-ffe9-4027-969f-dad511cc00f6.json]: Done.

**Review by tnorling — APPROVED** [review]
